authenticator/server/authenticator.go

```diff
@@ -193,6 +194,30 @@ func (a *authenticator) GetPGSHA256Hash(ctx context.Context, req *pb.PGSHA256Has
 	return &pb.PGSHA256Response{Cproof: cproof, Sproof: sproof}, nil
 }
 
+func (a *authenticator) GetMYSQLSHA1Hash(ctx context.Context, req *pb.MYSQLSHA1HashRequest) (*pb.MYSQLSHA1Response, error) {
+	a.incrementRequestCount()
+
+	// Return early if we didn't get a valid salt.
+	salt := req.GetSalt()
+	if len(salt) != 20 {
+		msg := fmt.Sprintf("expected salt to be 20 bytes long, but got %d bytes", len(salt))
+		log.Error(msg)
+		return nil, status.Errorf(codes.InvalidArgument, msg)
```

Comment:
Since there's a logging layer now that handles logging, could we change this from:
```
msg := fmt.Sprintf("expected salt to be 20 bytes long, but got %d bytes", len(salt))
log.Error(msg)
return nil, status.Errorf(codes.InvalidArgument, msg)
```
To:
```
return nil, status.Errorf(codes.InvalidArgument, fmt.Sprintf("expected salt to be 20 bytes long, but got %d bytes", len(salt)))
```

There are two benefits - 1. when it's logged, it will use the request logger with the request ID rather than the static logger (`log.Errorf`) that doesn't have the request ID, and 2. it keeps the code simpler because the logging is handled elsewhere.